backend/controllers/assetsPostingController.js

```diff
@@ -0,0 +1,103 @@
+const { mongo, default: mongoose } = require('mongoose');
+const AssetPosting = require('../models/AssetPostingModel');
+
+// get all asset postings
+const getAssetPostings = async (req, res) => {
+    const assetPostings = await AssetPosting.find({});
+    res.status(200).json(assetPostings);
+}
+
+// get an asset posting by ID
+const getAssetPostingById = async (req, res) => {
+    const { id } = req.params
+
+    if (!mongoose.Types.ObjectId.isValid(id)) {
+        return res.status(404).send('Asset posting not found')
+    }
+
+    const assetPosting = await AssetPosting.findById(id);
+
+    if (!AssetPosting) {
+        return res.status(404).send('Asset posting not found')
+    }
+
+    res.status(200).json(assetPosting);
+}
+
+// create a new asset posting
+const createAssetPosting = async (req, res) => {
+    const { company, assetType, location, availability, details, value, benefits } = req.body;
+
+    const newAssetPosting = new AssetPosting({
```

Comment:
should make this new object after you've done checking if the asset already exists